src/error.rs

```diff
@@ -0,0 +1,126 @@
+use thiserror::Error;
+pub type Result<T> = std::result::Result<T, Error>;
+
+#[derive(Error, Debug)]
+/// Defines all possible errors that can occur in this library
+pub enum Error {
```

Comment:
Is this also something that can be shared by the other uniffi-based bindings?

Reply:
Ideally the Rust API would use the same set  of errors as well. So I'd like to merge this into c2pa-rs and have these be the standard errors for all. Note that this needs that error review task have on the list.